### PR TITLE
feat: waveform visualizer

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary by CodeRabbit

* **New Features**
  * Interactive waveform scrubber added to the audio player with tap/drag seeking, adaptive layout, and accessibility hints; seamless fallback to progress bar.

* **Behavior**
  * Playback completion handling improved to reliably show final progress after completion.

* **Dependencies**
  * Added waveform extraction library for on-device waveform processing.

* **Tests**
  * Extensive unit and widget tests for extraction, caching, provider lifecycle, UI interactions, and edge cases.

* **Documentation**
  * Implementation plan and README updates detailing integration, caching, and rollout phases.

### CODE REVIEW

File: lib/features/speech/services/audio_waveform_service.dart [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+    } catch (error, stackTrace) {
+      _loggingService.captureException(
+        error,
+        domain: 'audio_waveform_service',
+        subDomain: 'cache_read',
+        stackTrace: stackTrace,
+      );
+      return null;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/speech/services/audio_waveform_service.dart:338]: no this is fine

File: lib/features/speech/services/audio_waveform_service.dart [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+      await _pruneCacheIfNeeded();
+    } catch (error, stackTrace) {
+      _loggingService.captureException(
+        error,
+        domain: 'audio_waveform_service',
+        subDomain: 'cache_write',
+        stackTrace: stackTrace,
+      );
+    }
+  }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/speech/services/audio_waveform_service.dart:356]: that's okay until it becomes a problem

File: lib/features/speech/services/audio_waveform_service.dart [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+      }
+    } catch (error, stackTrace) {
+      _loggingService.captureException(
+        error,
+        domain: 'audio_waveform_service',
+        subDomain: 'cache_prune',
+        stackTrace: stackTrace,
+      );
+    }
+  }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/speech/services/audio_waveform_service.dart:481]: moved to sync, I actually see a warning from analyzer otherwise to not use slow asynchronous IO

File: lib/features/speech/services/audio_waveform_service.dart [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+      final payload = _AudioWaveformCachePayload(
+        version: _waveformCacheVersion,
+        audioFileRelativePath: AudioUtils.getRelativeAudioPath(audio),
+        audioFileSizeBytes: stat.size,
+        audioFileModifiedAt: stat.modified.toUtc(),
+        audioDurationMs: waveform.duration.inMilliseconds,
+        bucketDurationMicros: bucketDuration.inMicroseconds,
+        amplitudes: normalized,
+        sampleCount: normalized.length,
+      );
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/speech/services/audio_waveform_service.dart:277]: fixed
